bika/lims/controlpanel/bika_analysisspecs.py

```diff
@@ -5,76 +5,131 @@
 from plone.app.layout.globals.interfaces import IViewView
+from Products.Archetypes import atapi
+from Products.ATContentTypes.content import schemata
+from zope.interface.declarations import implements
+
+
+# TODO: Separate content and view into own modules!
+
 
 class AnalysisSpecsView(BikaListingView):
     implements(IFolderContentsView, IViewView)
 
     def __init__(self, context, request):
         super(AnalysisSpecsView, self).__init__(context, request)
-        self.catalog = 'bika_setup_catalog'
-        self.contentFilter = {'portal_type': 'AnalysisSpec',
-                              'path': {'query':"/".join(self.context.getPhysicalPath()),
-                                       'level':0}}
-        self.sort_on = 'Title'
-        self.context_actions = {_('Add'):
-                                {'url': 'createObject?type_name=AnalysisSpec',
-                                 'icon': '++resource++bika.lims.images/add.png'}}
-        self.icon = self.portal_url + "/++resource++bika.lims.images/analysisspec_big.png"
+
+        self.catalog = "bika_setup_catalog"
+
+        self.contentFilter = {
+            "portal_type": "AnalysisSpec",
+            "sort_on": "sortable_title",
+            "sort_order": "ascending",
+            "path": {
+                "query": api.get_path(context),
+                "level": 0}
+        }
+
+        self.context_actions = {
+            _("Add"): {
+                "url": "createObject?type_name=AnalysisSpec",
+                "permission": "Add portal content",
+                "icon": "++resource++bika.lims.images/add.png"}
+        }
+
         self.title = self.context.translate(_("Analysis Specifications"))
         self.description = self.context.translate(_(
             "Set up the laboratory analysis service results specifications"))
+        self.icon = "{}/{}".format(
+            self.portal_url,
+            "/++resource++bika.lims.images/analysisspec_big.png"
+        )
+
         self.show_sort_column = False
         self.show_select_row = False
         self.show_select_column = True
         self.pagesize = 25
 
-        self.columns = {
-            'Title': {'title': _('Title'),
-                      'index': 'title',
-                      'replace_url': 'absolute_url'},
-            'SampleType': {'title': _('Sample Type'),
-                           'index': 'getSampleTypeTitle',
-                           'attr': 'getSampleType.Title',
-                           'replace_url': 'getSampleType.absolute_url'},
-        }
+        self.columns = collections.OrderedDict((
+            ("Title", {
+                "title": _("Analysis Specification"),
+                "index": "sortable_title"}),
+            ("SampleType", {
+                "title": _("Sample Type"),
+                "index": "getSampleTypeTitle",
+                "attr": "getSampleType.Title",
+                "replace_url": "getSampleType.absolute_url"}),
```

Comment:
Since you are managing the SampleType cell rendering in the folderitem function, there is no need of these two params (`attr` and `replace_url`) here. Otherwise, remove `folderitem` function, leave these two params for SampleType as they are and add the same mechanism for Title column